libs/klv/src/main/java/org/codice/alliance/libs/klv/SecurityClassificationKlvProcessor.java

```diff
@@ -13,11 +13,83 @@
  */
 package org.codice.alliance.libs.klv;
 
+import java.util.Collections;
+import java.util.Map;
+
+import org.codice.alliance.catalog.core.internal.api.classification.SecurityClassificationService;
 import org.codice.alliance.libs.stanag4609.Stanag4609TransportStreamParser;
 
-public class SecurityClassificationKlvProcessor extends DistinctSingleKlvProcessor {
-    public SecurityClassificationKlvProcessor() {
-        super(AttributeNameConstants.SECURITY_CLASSIFICATION,
-                Stanag4609TransportStreamParser.SECURITY_CLASSIFICATION);
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+/**
+ * Classification mapping defined by http://www.gwg.nga.mil/misb//docs/standards/ST0102.7.pdf
+ * <p>
+ * 1 ==> unclassified
+ * 2 ==> restricted
+ * 3 ==> confidential
+ * 4 ==> secret
+ * 5 ==> top secret
+ */
+public class SecurityClassificationKlvProcessor extends MultipleFieldKlvProcessor {
+
+    private Map<Short, String> codeToSecurityClassification;
+
+    private String defaultSecurityClassification;
+
+    private SecurityClassificationService securityClassificationService;
+
+    /**
+     * @param codeToSecurityClassification  map of stanag security codes to classification strings
+     * @param defaultSecurityClassification classification string to be used if the map does not contain an entry for the stanag security code
+     */
+    public SecurityClassificationKlvProcessor(
+            SecurityClassificationService securityClassificationService,
+            Map<Short, String> codeToSecurityClassification, String defaultSecurityClassification) {
+        super(Collections.singletonList(Stanag4609TransportStreamParser.SECURITY_CLASSIFICATION));
+        this.securityClassificationService = securityClassificationService;
+        this.codeToSecurityClassification = codeToSecurityClassification;
+        this.defaultSecurityClassification = defaultSecurityClassification;
+    }
+
+    public void setSecurityClassification(Short code, String classification) {
+        codeToSecurityClassification.put(code, classification);
+    }
+
+    public void setDefaultSecurityClassification(String classification) {
+        defaultSecurityClassification = classification;
+    }
+
+    @Override
+    public void accept(Visitor visitor) {
+        visitor.visit(this);
+    }
+
+    @Override
+    protected void doProcess(Attribute attribute, Metacard metacard) {
+
+        attribute.getValues()
+                .stream()
+                .filter(Short.class::isInstance)
+                .map(Short.class::cast)
+                .map(this::codeToClassification)
```

Comment:
We know that it's a classification attribute because this class looks for the data that came from Stanag4609TransportStreamParser.SECURITY_CLASSIFICATION. If you like, we can talk offline about the architecture and how it should probably be updated to make this more clear.